app/models/exchange_rates/xe_api.rb

```diff
@@ -0,0 +1,33 @@
+module ExchangeRates
+  class XeApi
+    class XeApiError < StandardError; end
+
+    def initialize(date: Time.zone.today)
+      @date = date.strftime('%Y-%m-%d')
+    end
+
+    def get_all_historic_rates
+      send_request("/v1/historic_rate.json/?from=GBP&date=#{@date}&to=*&amount=1")
+    end
+
+    private
+
+    def send_request(url)
+      uri = URI(TradeTariffBackend.xe_api_url + url)
+
+      client = Faraday.new(uri.host) do |conn|
+        conn.request :basic_auth, TradeTariffBackend.xe_api_username, TradeTariffBackend.xe_api_password
+      end
+
+      response = client.get(uri)
+
+      if response.success?
```

Comment:
If I am not wrong, Faraday does raise exceptions if the HTTP response is not a success (response outside 200..299 range). So I think we don't need manual check for success.

Reply:
I think this depends on middleware

Reply:
https://github.com/trade-tariff/trade-tariff-frontend/blob/main/app/services/client_builder.rb#L52